RFC-0010-jdk-upgrade.md

```diff
@@ -0,0 +1,293 @@
+# **RFC 10 for Presto**
+
+## JDK Upgrade for Presto
+
+Proposers
+
+* ZacBlanco
+* imjalpreet
+* aaneja
+
+
+## Summary
+
+This RFC lists out the steps and process for upgrading the build and runtime JDK version used in Presto to a more recent LTS release like 17 and 21  
+As of today (2025-01-24) the version we use is JDK 8
+
+
+## Background
+
+The Java community has slowly but surely been moving to JDK versions beyond JDK 8  
+
+Some of our dependencies, e.g [Jetty](https://github.com/jetty/jetty.project/) have already moved to JDK 17, and declared prior versions to be out-of-support(See https://github.com/jetty/jetty.project/issues/10485. The latest supported version is Jetty 12)
+
+Not using modern versions of our dependencies opens Presto installations to security risks through CVEs, which hurts adoption of Presto
+
+See this [Prestocon 2024 presentation](https://www.youtube.com/watch?v=K6CjuPZtqW4&list=PLJVeO1NMmyqW_qoMMEyVq-wW8lBSxA7hP&index=10) for a deep-dive motivating the issue further (Slides [here TODO]())
+
+
+## Proposed Migration Plan  
+
+To migrate to newer versions of the JDK, we propose a phased approach -   
+
+![slide from Prestocon 2024 presentation](RFC-0010/jdk_upgrade_timeline.png)
+
+The salient points - 
+
+ - During the upgrade, the project will be built with two JDK versions , the current and the new target version (e.g JDK 8 and 17) simultaneously
+ - We will make use of our CI pipelines to run the full breadth of unit, integration and end-to-end product tests with every new JDK version
+ - Users will be invited to test new builds with their production workloads to report any issues observed around correctness, performance and stability
+ - As blocking issues shake out and get resolved, we will start deprecating support for older JDK versions
```

Comment:
I think we should be specific about timelines (e.g. 1 release or 2 release version window where we attempt dual support before dropping support for the older JDK)

Reply:
Okay, added